osbuild/buildroot.py

```diff
@@ -194,6 +194,22 @@ def run(self, argv, binds=None, readonly_binds=None):
         cmd += ["--", f"/run/osbuild/lib/runners/{self._runner}"]
         cmd += argv
 
-        return subprocess.run(cmd,
-                              check=False,
-                              stdin=subprocess.DEVNULL)
+        proc = subprocess.Popen(cmd,
+                                bufsize=0,
+                                stdin=subprocess.DEVNULL,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.STDOUT,
+                                encoding="utf-8",
+                                close_fds=True)
+
+        while True:
+            txt = proc.stdout.read(4096)
+            if not txt:
+                break
+
+            monitor.log(txt)
+
+        txt, _ = proc.communicate()
+        monitor.log(txt)
```

Comment:
What is the reason for the `communicate()` call? `stdin` is `DEVNULL`, `stderr` is on `STDOUT`, and `stdout` was already drained. Will this actually do anything? Or is it just a "safer" call to `proc.wait()`? That is, we do not expect this to return anything, but it also will not do any harm, right?

Reply:
> That is, we do not expect this to return anything, but it also will not do any harm, right?

Basically this, but it is also a shortcut for `proc.wait()` and `proc.stdout.close()`; without the latter you get nasty `pipeline.py:301: ResourceWarning: unclosed file <_io.TextIOWrapper name=17 encoding='utf-8'>` warnings.